packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,239 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { Button, ButtonAppearance } from "../button";
+import {
+    ActionToggleAppearance,
+    ActionToggleHandledProps,
+    ActionToggleProps,
+    ActionToggleUnhandledProps,
+} from "./action-toggle.props";
+import { actionToggleButtonOverrides } from "@microsoft/fast-components-styles-msft";
+import { isNullOrWhiteSpace } from "../../../fast-web-utilities/dist";
+import { isNullOrUndefined } from "util";
+
+export interface ActionToggleState {
+    selected: boolean;
+}
+
+class ActionToggle extends Foundation<
+    ActionToggleHandledProps,
+    ActionToggleUnhandledProps,
+    ActionToggleState
+> {
+    public static displayName: string = "ActionToggle";
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: ActionToggleProps,
+        prevState: ActionToggleState
+    ): null | Partial<ActionToggleState> {
+        if (
+            typeof nextProps.selected === "boolean" &&
+            nextProps.selected !== prevState.selected
+        ) {
+            return {
+                selected: nextProps.selected,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<ActionToggleHandledProps> = {
+        appearance: void 0,
+        managedClasses: void 0,
+        disabled: void 0,
+        selected: void 0,
+        selectedGlyph: void 0,
+        unselectedGlyph: void 0,
+        selectedText: void 0,
+        unselectedText: void 0,
+        selectedARIALabel: void 0,
+        unselectedARIALabel: void 0,
+    };
+
+    /**
+     * Define constructor
+     */
+    constructor(props: ActionToggleProps) {
+        super(props);
+
+        this.state = {
+            selected: this.props.selected || false,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        return (
+            <Button
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                disabled={this.props.disabled}
+                onClick={this.handleToggleChange}
+                aria-label={this.renderARIALabel()}
+                appearance={
+                    ButtonAppearance[ActionToggleAppearance[this.props.appearance]]
+                }
+                jssStyleSheet={actionToggleButtonOverrides}
+            >
+                {this.renderGlyph()}
+                {this.renderLabel()}
```

Comment:
What is the difference between label and children? I think this should really just support children, which leads to the question as to whether this should be a controlled component only

Reply:
my thoughts on this in your other thread

Reply:
after discussion we ended up not supporting children and content can be a react node rather than just a string